src/pandantic/validators/pandas.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import math
+import os
+from typing import Any, Optional
+
+import pandas as pd
+from multiprocess import (  # type:ignore # pylint: disable=no-name-in-module
+    Process,
+    Queue,
+)
+
+from pandantic.types import SchemaTypes
+from pandantic.validators.baseclass import BaseValidator
+
+
+class PandasValidator(BaseValidator):
+    def __init__(self, schema: SchemaTypes):
+        self.schema = schema
+
+    def validate(
+        self,
+        dataframe: pd.DataFrame,
+        errors: str = "raise",
+        context: Optional[dict[str, Any]] = None,  # pylint: disable=consider-alternative-union-syntax,useless-suppression
+        n_jobs: int = 1,
+        verbose: bool = True,
+    ) -> pd.DataFrame:
+        """Validate a DataFrame using the schema defined in the Pydantic model.
+
+        Args:
+            dataframe (pd.DataFrame): The DataFrame to validate.
+            errors (str, optional): How to handle validation errors. Defaults to "raise".
+            context (Optional[dict[str, Any], None], optional): The context to use for validation.
+            n_jobs (int, optional): The number of processes to use for validation. Defaults to 1.
+            verbose (bool, optional): Whether to log validation errors. Defaults to True.
+
+        Returns:
+            pd.DataFrame: The DataFrame with valid rows in case of errors="filter".
+        """
+        errors_index = []
+        logging.debug("Amount of available cores: %s", os.cpu_count())
+
+        dataframe = dataframe.copy()
```

Comment:
Let's see if we can avoid doing any copying. With large dataframes that could pose a memory risk, especially if people want to deploy this in async services where it could be running in parallel with other dataframes.

Reply:
Agree this is a waste of resources but I am not sure how to move forward without copying and not mutation the original dataframe. Maybe with the new future methods (validate & filter) we could look into this challenge, maybe creating an index table over copying the dataframe?

Reply:
We can avoid mutating the original dataframe by not adding columns to it, but just reading values from it. Since we convert everything into a dict at one point, we might as well incur that cost easy and just use the dicts such that we only copy once (the dicts), versus technically three times (df1, df2, dict version).

**I just went ahead and implemented this!** Small tweak where we use `df.to_dict("Index")` which produces a `dict[index, dict[column, value]`. Such that we can iterate over  the .items() of the dict and get the index (without making a new column) and the data for validation.